control-plane-operator/hostedclusterconfigoperator/controllers/inplaceupgrader/inplaceupgrader.go

```diff
@@ -455,9 +456,14 @@ func (r *Reconciler) reconcileInPlaceUpgradeManifests(ctx context.Context, hoste
 func (r *Reconciler) reconcileUpgradeConfigmap(ctx context.Context, configmap *corev1.ConfigMap, targetConfigVersionHash string, payload []byte) error {
 	log := ctrl.LoggerFrom(ctx)
 
-	// TODO (jerzhang): should probably parse the data here to reduce size/compress
+	// Base64-encode and gzip the payload to allow larger overall payload sizes.
+	compressedPayload, err := util.CompressAndEncode(payload)
```

Comment:
@yuqi-zhang how's this not breaking the ability of the upgrader mco pod to read this content?